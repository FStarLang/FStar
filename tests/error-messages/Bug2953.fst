module Bug2953

open FStar.Reflection
open FStar.Tactics.Effect

let f (t : term) : Tac (term) = t

noeq
type ty =  | A : bool -> term -> ty

[@@expect_failure [236]]
let open_view (tv:ty) : Tac unit =
  match tv with
  | A recf body when recf=false ->
    let _ : term = f body in
    ()
