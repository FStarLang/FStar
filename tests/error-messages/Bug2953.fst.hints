[ "oøo\u001c¹)\u0015µª\u001bùdÙSp\u0012", [] ]