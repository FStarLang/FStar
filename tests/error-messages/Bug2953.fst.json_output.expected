>> Got issues: [
{"msg":["When clauses are not yet supported in --verify mode; they will be some day"],"level":"Error","range":{"def":{"file_name":"Bug2953.fst","start_pos":{"line":14,"col":21},"end_pos":{"line":14,"col":31}},"use":{"file_name":"Bug2953.fst","start_pos":{"line":14,"col":21},"end_pos":{"line":14,"col":31}}},"number":236,"ctx":["While typechecking the top-level declaration `let open_view`","While typechecking the top-level declaration `[@@expect_failure] let open_view`"]}
>>]
